### PR TITLE
[inductor] maintain eager stride for outputs

### PR DESCRIPTION
**Pull request by shunting314**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #102376
* #99773

This PR make sure we don't change the layout for original fwd graph's outputs. This means 2 things
- for inference, all graph output's layout are kept the same as eager
- for training, we keep the eager strides for the original model outputs returned in the fwd graph. We may still change the layout for saved activations.

This fixes the accuracy check fail for moco training.

Also confirmed that we still get all the perf wins from layout optimization [dashboard link](https://hud.pytorch.org/benchmark/compilers?startTime=Wed%2C%2024%20May%202023%2018%3A48%3A56%20GMT&stopTime=Wed%2C%2031%20May%202023%2018%3A48%3A56%20GMT&granularity=hour&suite=torchbench&mode=training&dtype=amp&lBranch=shunting-layout-opt-19&lCommit=baa797fc100688dfb044fbcbdebcfd2591710f78&rBranch=main&rCommit=999bae0f54108ffc5b7cf2524a02a83901554b16)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @anijain2305

### CODE REVIEW

**Comment by shunting314** [comment]
Combined this into https://github.com/pytorch/pytorch/pull/99773
